eclipse-collections-code-generator/src/main/resources/impl/map/mutable/primitiveObjectHashMap.stg

```diff
@@ -1071,6 +1072,16 @@ public class <name>ObjectHashMap\<V> implements Mutable<name>ObjectMap\<V>, Exte
         return null;
     }
 
+    public Optional\<V> detectOptional(Predicate\<? super V> predicate)
+    {
+        return Optional.ofNullable(this.detect(predicate));
```

Comment:
It's a good point, Optional.of(null) throws, unlike Scala's Option, so it cannot represent finding and returning null, or calling Map.get() and finding that the key was present but the value was null. So for backwards compatibility I guess I'm wrong, and we cannot delegate from detect to detectOptional.